packages/fx-core/src/component/generator/templates/templateNames.ts

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import {
+  ApiMessageExtensionAuthOptions,
+  CapabilityOptions,
+  CustomCopilotAssistantOptions,
+  CustomCopilotRagOptions,
+  MeArchitectureOptions,
+  NotificationTriggerOptions,
+} from "../../../question";
+
+export enum TemplateNames {
+  Tab = "non-sso-tab",
+  SsoTab = "sso-tab",
+  SsoTabObo = "sso-tab-with-obo-flow",
+  TabSSR = "non-sso-tab-ssr",
+  SsoTabSSR = "sso-tab-ssr",
+  DashboardTab = "dashboard-tab",
+  NotificationRestify = "notification-restify",
+  NotificationWebApi = "notification-webapi",
+  NotificationHttpTrigger = "notification-http-trigger",
+  NotificationHttpTriggerIsolated = "notification-http-trigger-isolated",
+  NotificationTimerTrigger = "notification-timer-trigger",
+  NotificationTimerTriggerIsolated = "notification-timer-trigger-isolated",
+  NotificationHttpTimerTrigger = "notification-http-timer-trigger",
+  NotificationHttpTimerTriggerIsolated = "notification-http-timer-trigger-isolated",
+  CommandAndResponse = "command-and-response",
+  Workflow = "workflow",
+  DefaultBot = "default-bot",
+  MessageExtension = "message-extension",
+  MessageExtensionAction = "message-extension-action",
+  MessageExtensionSearch = "message-extension-search",
+  MessageExtensionCopilot = "message-extension-copilot",
+  M365MessageExtension = "m365-message-extension",
+  TabAndDefaultBot = "non-sso-tab-default-bot",
+  BotAndMessageExtension = "default-bot-message-extension",
+  LinkUnfurling = "link-unfurling",
+  AIBot = "ai-bot",
+  AIAssistantBot = "ai-assistant-bot",
+  ApiPluginFromScratch = "api-plugin-from-scratch",
+  CopilotPluginFromScratch = "copilot-plugin-from-scratch",
+  CopilotPluginFromScratchApiKey = "copilot-plugin-from-scratch-api-key",
+  ApiMessageExtensionSso = "api-message-extension-sso",
+  CustomCopilotBasic = "custom-copilot-basic",
+  CustomCopilotRagCustomize = "custom-copilot-rag-customize",
+  CustomCopilotRagAzureAISearch = "custom-copilot-rag-azure-ai-search",
+  CustomCopilotRagCustomApi = "custom-copilot-rag-custom-api",
+  CustomCopilotRagMicrosoft365 = "custom-copilot-rag-microsoft365",
+  CustomCopilotAssistantNew = "custom-copilot-assistant-new",
+  CustomCopilotAssistantAssistantsApi = "custom-copilot-assistant-assistants-api",
+}
+
+export const Feature2TemplateName = {
+  [`${CapabilityOptions.nonSsoTab().id}:undefined`]: TemplateNames.Tab,
```

Comment:
Look the format of key is like "${capability}:${option}", can we make it as a real map table to avoid string concatenation issue

Reply:
Yes. I will refine the map logic in next PR. This PR focus on the implementation of new template generator class.